src/ai/ai.cpp

```diff
@@ -667,114 +667,84 @@ void take_ai_decisions(sys::state& state) {
 	}
 }
 
-dcon::issue_option_id find_most_important_issue_option(sys::state& state, dcon::nation_id n) {
-	dcon::issue_option_id iss;
-	float max_support = 0.f;
-	state.world.for_each_issue_option([&](dcon::issue_option_id io) {
-		if(command::can_enact_issue(state, n, io)) {
-			float support = 0.f;
-			for(const auto poid : state.world.nation_get_province_ownership_as_nation(n)) {
-				for(auto plid : state.world.province_get_pop_location_as_province(poid.get_province())) {
-					support += state.world.pop_get_demographics(plid.get_pop(), pop_demographics::to_key(state, io)) * state.world.pop_get_militancy(plid.get_pop());
-				}
+float estimate_pop_party_support(sys::state& state, dcon::nation_id n, dcon::political_party_id pid) {
+	auto iid = state.world.political_party_get_ideology(pid);
+	/*float v = 0.f;
+	for(const auto poid : state.world.nation_get_province_ownership_as_nation(n)) {
+		for(auto plid : state.world.province_get_pop_location_as_province(poid.get_province())) {
+			float weigth = plid.get_pop().get_size() * 0.001f;
+			v += state.world.pop_get_demographics(plid.get_pop(), pop_demographics::to_key(state, iid)) * weigth;
+		}
+	}*/
+	return state.world.nation_get_demographics(n, demographics::to_key(state, iid));
 }
-			if(support > max_support) {
-				iss = io;
-				max_support = support;
-			}
-		}
-	});
-	return iss;
+
+bool ai_can_appoint_political_party(sys::state& state, dcon::nation_id n) {
+	if(!politics::can_appoint_ruling_party(state, n))
+		return false;
+	auto last_change = state.world.nation_get_ruling_party_last_appointed(n);
+	if(last_change && state.current_date < last_change + 365)
+		return false;
+	if(politics::is_election_ongoing(state, n))
+		return false;
+	// Do not appoint if we are a democracy!
+	if(politics::has_elections(state, n))
+		return false;
+	/*auto gov = state.world.nation_get_government_type(source);
+	auto new_ideology = state.world.political_party_get_ideology(p);
+	if((state.culture_definitions.governments[gov].ideologies_allowed & ::culture::to_bits(new_ideology)) == 0) {
+		return false;
+	}*/
+	return true;
 }
 
 void update_ai_econ_construction(sys::state& state) {
 	for(auto n : state.world.in_nation) {
 		// skip over: non ais, dead nations, and nations that aren't making money
-		if(n.get_is_player_controlled() || n.get_owned_province_count() == 0 || n.get_spending_level() < 1.0f || n.get_last_treasury() >= n.get_stockpiles(economy::money))
+		if(n.get_is_player_controlled() || n.get_owned_province_count() == 0)
 			continue;
+		//if(n.get_spending_level() < 1.0f || n.get_last_treasury() >= n.get_stockpiles(economy::money))
+		//	continue;
+
+		// buy stuff from the global market if we need it
+		state.world.for_each_commodity([&](dcon::commodity_id c) {
+			n.set_stockpile_targets(c, 10000.f);
```

Comment:
This will result in a lot of purchases that aren't necessary and it won't work as you expect. If you try to fill the stockpile from empty, it is almost certain all that demand won't be satisfied. So this will result in stockpiles flipping on and off AND starving factories and consumers of goods

Reply:
so i just reverse it? when demand is satisfied i buy it up, and when it is not, i sell it?